feat: document beta releases expectations (#5565)
expressjs/express: Release-Process.md

- This branch contains the commits accepted so far that implement the proposal
  in the tracking pull request.

> ### Beta releases

Reviewer: ```suggestion
### Pre-release Versions
```

Can we change this? So that we cover both the alpha and beta releases?

Alpha and Beta releases are made from a proposal branch. The version number should be
incremented to the next minor version with a `-beta` or `-alpha` suffix.
For example, if the next beta release is `5.0.1`, the beta release would be `5.0.1-beta.0`.

> The pre-releases should be considered unstable and not suitable for production use.

Reviewer 2: ```suggestion
The pre-releases are unstable and not suitable for production use.
```
